packages/fetch-engine/src/download.ts

```diff
@@ -294,37 +291,44 @@ async function binaryNeedsToBeDownloaded(
   // If there is no cache and the file doesn't exist, we for sure need to download it
   if (!targetExists) {
     debug(`file ${job.targetFilePath} does not exist and must be downloaded`)
+
     return true
   }
 
   // 3. If same platform, always check --version
```

Comment:
```suggestion
  // 3. If same platform, check --version and compare to expected version
  // If version don't match, force download
```
Maybe something like that?